File: IPRS.Server/Models/Notification.cs
// Models/Notification.cs
using System.ComponentModel.DataAnnotations;

namespace IPRS.Server.Models;

public class Notification
{
    [Key]
    public Guid Id { get; set; } = Guid.NewGuid();

    [Required, MaxLength(300)] 
    public string Message { get; set; } = string.Empty;
    public bool IsRead { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public Guid  UserId { get; set; }
    public Guid? RelatedRequestId { get; set; }
    
    // Relationships
    public User User { get; set; } = null!;
    public PurchaseRequest? RelatedRequest { get; set; }
}

File: iprs.client/src/schemas/notification.ts
import { z } from 'zod';

export const notificationResponseSchema = z.object({
  id: z.guid(),
  message: z.string(),
  isRead: z.boolean(),
  createdAt: z.iso.datetime(),
  relatedRequestId: z.guid(),
});



export const updateNotificationReadStatusSchema = z.object({
  isRead: z.boolean(),
});

